Makefile

```diff
@@ -101,3 +102,26 @@ release: test dist
 	git push --tags
 	open https://github.com/$(NAME)/releases/tag/v$(VERSION)
 	open dist
+
+.PHONY: generate-grpc
+generate-grpc:
```

Comment:
This make script depends on a few unstated things:

* the `protoc` binary
* the vendor/ directory (from `go mod vendor`?)
* the `github.com/grpc-ecosystem/grpc-gateway` package (so that `GATEWAY_SRC` resolves)
* i'm currently blocked on
  > protoc-gen-gogoslick: program not found or is not executable`

Which of those things can be resolved automatically in the Makefile using a dependency, and which need to be documented in the README?

Reply:
Regarding the protobuf dependencies in vendor, can I suggest we just copy the ones we need into the repo (e.g. https://github.com/hyperledger/burrow/tree/develop/protobuf). I realise this feels a bit icky but:

- It makes it much easier for other languages to consume our protobuf - particular javascript which until recently couldn't even handle multiple import roots. It's a pain for a python or C++ app to have to care about go tooling to get vendor
- Just because a Go dependency depends on a particular protobuf version it doesn't automatically mean we do (though it kind of does for grpc-gateway I suppose)
- These google protobuf types are very stable - we realistically will not have to worry about updating them any time soon

We can include a make target to copy them in to our GRPC as a middle ground...

Reply:
Okay, I was able to remove dependency on `vendor/`. Regarding the dependency on `protoc` and the other binaries, a decision can be made between:

- document the dependencies in the README
- use the [znly/protoc](https://hub.docker.com/r/znly/protoc) docker image

The first one has the advantage of owning our dependency, kinda, and being able to always be on the latest version, but they're a few additional stuff to install. The Docker image route requires no additional installs, but it's a dependency on an external Docker image and we don't own the update cycle.

Opinions?

Reply:
I'm inclined to use a Docker image with all of the tooling necessary to generate these files. That seems consistent with how this Makefile currently relies on Docker when it needs a specific build environment.

We can always swap out one image for another (even our own) if need arises.

Reply:
Coincidentally, I made the same decision for the same reason last weekend :) Working on the necessary changes now

Reply:
Done!